perllib/FixMyStreet/Cobrand/HighwaysEngland.pm

```diff
@@ -159,28 +159,50 @@ sub munge_report_new_contacts {
 }
 
 sub national_highways_cleaning_groups {
-    my ($self, $category_groups, $c) = @_;
-    my @litter_cats;
-    my %cleaning_cats = map { $_ => 1 } @{ $self->_cleaning_categories };
-    for my $cat (@$category_groups) {
-        if ($cat->{name} && ($cat->{name} =~ $self->council_name)) {
-            for my $cat_item (@{ $cat->{categories}}) {
-                my $clean_name = $cat_item->category_display;
-                $clean_name =~ s/ \(NH\)//;
-                $cat_item->set_extra_metadata('display_name', $clean_name);
-            }
-            if ($c->stash->{report_new_is_on_he_road_not_litter}) {
-                @{ $cat->{categories} } = grep {$_->category_display !~ /Flytipping|Litter/} @{ $cat->{categories} };
-            }
-        }
+    my ($self, $category_groups) = @_;
+
+    # We only care if there is a National Highways group
+    my ($nh_group) = grep { ($_->{name} || '') eq $self->council_name } @$category_groups;
+    return unless $nh_group;
+
+    my $c = $self->{c};
+    my $not_he_litter = $c->stash->{report_new_is_on_he_road_not_litter};
+    if (!defined $not_he_litter) {
+        # We delayed working it out because every /around page calls
+        # report_new_is_on_he_road but do it now we need to know
+        my ($x, $y) = ($c->stash->{longitude}, $c->stash->{latitude});
+        $not_he_litter = $self->_report_new_is_on_he_road_not_litter($x, $y);
+    }
 
-        if (!$cat->{name} && $cleaning_cats{$cat->category_display} && $c->stash->{report_new_is_on_he_road_not_litter}) {
-            push @litter_cats, $cat;
-        }
+    for my $cat_item (@{$nh_group->{categories}}) {
+        my $clean_name = $cat_item->category_display;
+        $clean_name =~ s/ \(NH\)//;
+        $cat_item->set_extra_metadata('display_name', $clean_name);
     }
+
+    # Don't change anything else unless we're on a HE non-litter road
+    return unless $not_he_litter;
+
+    # NH do not want flytipping or litter reports on these roads, take them out
+    @{$nh_group->{categories}} = grep { $_->category_display !~ /Flytipping|Litter/ } @{$nh_group->{categories}};
+
+    # Put any council street cleaning categories we can find into the NH group,
```

Comment:
This is expanded on what was there, to also cover cleaning subcategories and cleaning category groups.